cmd/cluster/azure/create.go

```diff
@@ -101,8 +102,7 @@ func (o *RawCreateOptions) Validate(ctx context.Context, opts *core.CreateOption
 		},
 	}
 
-	var err error
-	validOpts.ValidatedAzurePlatformCreateOptions, err = o.NodePoolOpts.Validate()
+	validOpts.ValidatedAzurePlatformCreateOptions, err = o.NodePoolOpts.Validate(ctx, opts.Arch, opts.Name, opts.Namespace)
```

Comment:
can we instead pass down the whole opts *core.CreateOptions resource so we keep signatures consistent between platforms and we can then put an interface atop to keep implementations aligned and easier to follow and maintain

Reply:
If we want to do this, can we do it in a follow up ticket/PR?

The problem I ran into is the structs are different between cmd/cluster/azure/create.go (opts *core.CreateOptions) and when this same function is called in cmd/nodepool/azure/create.go (coreOpts *core.CreateNodePoolOptions). I think we would want to bring those two structs into alignment or just make one struct and use it in both places. To me, making that change is out of scope for this PR.

Reply:
That can be solved with an interface that force both to implement GetClusterName, GetArch and GetNamespace. Can we at a minimum at a TODO and link a jira ticket capturing all this?

aside form this PR why isn't AzureMarketPlaceImageInfo a struct of AzurePlatformCreateOptions?